### ISSUE TITLE
Plan to squat names of Fedora packages

### ISSUE THREAD

**Issue by encukou** [pypa/pypi-support#355]
*TL;DR: I want to squat ~250 names on PyPI before opening discussions on how to handle them properly.*

(This may be out of place, but I don't know of a better way to reach PyPI moderators. Please let me know if I should post elsewhere.)

Hello,
As PyPI has become the de-facto canonical namespace for freely redistributable Python packages, I would like to encourage maintainers of packages in the [Fedora](https://getfedora.org/) Linux distro to synchronize their [Distribution Package](https://packaging.python.org/glossary/) names with PyPI – effectively, to merge Fedora's namespace into PyPI's.

This was not yet discussed either on the Fedora side nor on the PyPA one. I plan to kick off those two discussions soon.

But just before kicking off the discussions, I'd like to register distributon names from Fedora that aren't on PyPI yet, so that:
* good faith actors are advised that the name is taken in some sense (though not in the [PEP 541](https://www.python.org/dev/peps/pep-0541/#invalid-projects) sense), so using the name for a different purpose might confuse users, and
* bad-faith actors can't snatch the names that easily.

As PyPI [doesn't have a dedicated mechanism to do this](https://discuss.python.org/t/pep-541-should-name-squatting-be-actively-discouraged/2407/9?u=encukou), I plan to squat about 250 package names on PyPI. That is, upload non-installable packages with an informative description inspired by the existing [`microsoft`](https://pypi.org/project/microsoft/) and [`fedora`](https://pypi.org/project/fedora/) name squats.
I know well that this is [against current regulations](https://www.python.org/dev/peps/pep-0541/#invalid-projects). I'm of course prepared to hand ownership to the software authors and/or maintainers when they are ready to publish to PyPI. I'm also prepared to delete all the projects if the discussions don't go my way. I plan to use a version like to `0.0.0.1.dev0` to avoid polluting that space.

Please let me know if you believe I should not attempt this, either here or privately at pviktori@redhat.com.
If there is automation to prevent me from doing this, I would be grateful to get an exception for the PyPI account `pviktori` (which is currently empty). Provided you aren't opposed to the idea, of course.

cc @hroncok

**Comment by di** [comment]
Hi @encukou, thanks for reaching out beforehand.

One thing we could do instead is temporarily mark these packages as unregistrable. Once the Fedora community has come up with a plan, we can coordinate w/ you to release them for registration.

Also, for future reference, this is an appropriate place for this issue, but you can also email us at admin@pypi.org if you ever need to.

**Comment by hroncok** [comment]
I'd consider marking them unregistrable much cleaner. Would it give a rarionale to the user who attempts to register it, or at least point them to this tracker?

Thank you Dustin.

**Comment by encukou** [comment]
@di: Thank you! I've sent an e-mail with the list.

@hroncok: Yes. Here's what I get when I try to upload `math`:
```
$ twine upload -r testpypi dist/math-0.tar.gz 
Uploading distributions to https://test.pypi.org/legacy/
Uploading math-0.tar.gz
100%|███…|
NOTE: Try --verbose to see response content.
HTTPError: 400 Client Error: The name 'math' isn't allowed (conflict with Python Standard Library module name). See https://test.pypi.org/help/#project-name for more information. for url: https://test.pypi.org/legacy/
```

**Comment by dstufft** [comment]
To be clear, the standard library handling is special. For our standard blacklist functionality the error message is much more generic. It'll basically just omit the parenthetical.

**Comment by encukou** [comment]
That's not ideal, but I'm only looking for an interim solution here.
Thanks for pointing it out, though; I'll note it in the Fedora discussion.

**Comment by ewdurbin** [comment]
@di @dstufft The requested names have been added to the prohibited project names list with comment `fedora distribution`.

**Comment by encukou** [comment]
Thank you very much!
The discussion is at https://discuss.python.org/t/pypi-as-a-project-repository-vs-name-registry-a-k-a-pypi-namesquatting-e-g-for-fedora-packages/4045?u=encukou .

**Comment by encukou** [comment]
These names are reserved mainly to prevent people from accidentally picking a name that is already used in Fedora.

If you maintain a long-standing project whose name was reserved for Fedora, and want to upload it to PyPI, please let me (or the PyPI admins) know so it can be unblocked for you.

**Comment by hroncok** [comment]
See also https://github.com/pypa/pypi-support/issues/529

**Comment by dimpase** [comment]
Please ublock `sagetex`. #1516 is a result of this, as Fedora distributes SageMath, sagetex is a SageMath package, so they have actually blocked us (SageMath) from using this name!

**Comment by encukou** [comment]
The name is reserved for *you*. The name reservation is intended to block people who'd want to use the name for a *different* project.
Unfortunately with this many packages, it's not feasible to contact the owner of each project and get them to grab the name on PyPI.

Sorry for the wait; hopefully a PyPI admin can unblock the name shortly.

**Comment by encukou** [comment]
I adjusted the title & first message, hopefully the situation is now clearer.

**Comment by dimpase** [comment]
It took me a while to find this issue, in view of #1516. `twine`'s message is rather unhelpful in this respect, as these `reserved` names are not in any available in a normal way list. 

So one has to trust a search engine, and then do not overlook the hidden under "list of packages" item.
